clearnode/rpc_router_private.go

```diff
@@ -276,8 +306,19 @@ func (r *RPCRouter) HandleTransfer(c *RPCContext) {
 
 	var respTransactions []TransactionResponse
 	err = r.DB.Transaction(func(tx *gorm.DB) error {
-		if wallet := GetWalletBySigner(fromWallet); wallet != "" {
-			fromWallet = wallet
+		// Determine if wallet signed; only apply session-key limits if wallet did NOT sign.
```

Comment:
The "determine if wallet signed" logic is used in many places. Is it possible to extract it into a helper function somehow?